pymongo/uri_parser.py

```diff
@@ -129,6 +130,26 @@ def parse_host(entity, default_port=DEFAULT_PORT):
     return host.lower(), port
 
 
+def _parse_options(opts, delim):
+    """Helper method for split_options which creates the options dict.
+    Also handles the creation of a list for the URI tag_sets/
+    readpreferencetags portion and the use of the tlsInsecure option."""
+    options = {}
+    for opt in opts.split(delim):
+        key, value = opt.split("=")
+        optname = str(key).lower()
+        if optname == 'readpreferencetags':
+            options.setdefault(optname, []).append(value)
+        else:
+            if optname == 'tlsinsecure':
+                options['tlsAllowInvalidCertificates'] = value
+                options['tlsAllowInvalidHostnames'] = value
+            if key in options:
+                warnings.warn("Duplicate URI option '%s'." % (str(key),))
```

Comment:
If the application specifies tlsInsecure and one or both of the tlsAllow* options you added above you're going to trigger this warning unnecessarily.

Reply:
We definitely want _some_ warning to be triggered when `tlsInsecure` is specified along with one of the other `tlsAllow*` options as evidenced by this [spec test](https://github.com/mongodb/specifications/pull/414/files#diff-6e7064dc748fe47120d108a28f561240R87). I agree that the duplicate option warning is not very appropriate in this situation, but I thought it was simpler than introducing a new warning... What kind of warning would you like to see @behackett ?